cmd/generic-handlers.go

```diff
@@ -213,7 +213,7 @@ func setPrivateBucketHandler(h http.Handler) http.Handler {
 func (h minioPrivateBucketHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// For all non browser requests, reject access to 'minioReservedBucketPath'.
 	bucketName, _ := urlPath2BucketObjectName(r.URL)
-	if !guessIsBrowserReq(r) && isMinioReservedBucket(bucketName) && isMinioMetaBucket(bucketName) {
+	if !guessIsBrowserReq(r) && isMinioReservedBucket(bucketName) || isMinioMetaBucket(bucketName) {
```

Comment:
Due to `&&` being higher precedence than `||` this will reject requests to any minio-meta-bucket regardless of it being a browser request. I am not sure if that is intended to fix the bug that is being addressed here (if it is, the comment needs to be updated too). It looks like the intended conditional is `!guessIsBrowserReq(r) && (isMinioReservedBucket(bucketName) || isMinioMetaBucket(bucketName))`.

Reply:
You are right, let me change that. @donatello